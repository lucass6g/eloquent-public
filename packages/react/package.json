{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
