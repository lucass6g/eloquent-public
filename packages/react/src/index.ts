export {Switch, Case, Default} from './components/conditional-switch'