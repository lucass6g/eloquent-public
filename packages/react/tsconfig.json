{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
