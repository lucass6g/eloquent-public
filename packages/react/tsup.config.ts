import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})