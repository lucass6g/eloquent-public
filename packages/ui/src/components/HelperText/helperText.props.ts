export interface RootHelperTextProps
  extends React.HTMLAttributes<HTMLDivElement> {}

export interface LabelHelperTextProps extends React.HTMLAttributes<HTMLSpanElement> {}