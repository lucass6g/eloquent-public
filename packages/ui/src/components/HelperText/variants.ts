import { tv } from "tailwind-variants";

export const helperTextVariants = tv({
    slots: {
        root: "flex flex-col gap-2",
        label: "text-sm text-neutral-500"
    }
})