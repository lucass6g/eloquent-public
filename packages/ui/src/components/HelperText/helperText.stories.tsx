import { Meta, StoryObj } from '@storybook/react';
import { HelperText } from '.';

const meta = {
component: () => {
    return (
        <HelperText.Root>
            <HelperText.Label>Seu texto de apoio.</HelperText.Label>
        </HelperText.Root>
    )
},
  title: 'HelperText',
  parameters: {
    layout: 'centered'
},
} satisfies Meta<{}>;

export default meta;
type Story = StoryObj<typeof meta>;

export const HelperTextDefault: Story = {
name: 'HelperText',
args: {},
render: () => {
    return (
        <HelperText.Root>
            <HelperText.Label>Seu texto de apoio.</HelperText.Label>
        </HelperText.Root>
    )
  },
};