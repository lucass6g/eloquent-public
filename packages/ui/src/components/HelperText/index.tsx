import React from "react";
import { cn } from "@eloquent/styles";
import { LabelHelperTextProps, RootHelperTextProps } from "./helperText.props";
import { helperTextVariants } from "./variants";

const {root, label} = helperTextVariants()

const RootHelperText = React.forwardRef<HTMLDivElement, RootHelperTextProps>(
    function RootHelperText({className, children, ...props}, ref){
        return(
            <div className={cn(root(), className)} ref={ref} {...props}>
                {children}
            </div>
        )
    }
)

const LabelHelperText = React.forwardRef<HTMLSpanElement, LabelHelperTextProps>(
    function LabelHelperText({className, children, ...props}, ref){
        return(
            <span className={cn(label(), className)} ref={ref} {...props}>
                {children}
            </span>
        )
    }
)

export const HelperText = {
    Root: RootHelperText,
    Label: LabelHelperText
}