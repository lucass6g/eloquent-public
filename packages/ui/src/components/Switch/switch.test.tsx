import { render, screen, waitFor } from "@testing-library/react";
import { it, describe, expect, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import { Switch } from "..";
import { HelperText } from "../HelperText";

describe("Switch", () => {
  it("should match the snapshot", () => {
    const component = render(<Switch.Root id="switch" />);

    expect(component.baseElement).toMatchSnapshot();
  });

  it("should check correctly", async () => {
    const switchFn = vi.fn();

    render(<Switch.Button defaultChecked={false} onCheckedChange={switchFn} />);

    const button = screen.getByRole("switch");
    expect(button).not.toBeChecked();

    await waitFor(() => userEvent.click(button));
    expect(button).toBeChecked();

    expect(switchFn).toHaveBeenCalled();
  });

  it("should uncheck correctly", async () => {
    const switchFn = vi.fn();

    render(<Switch.Button defaultChecked={true} onCheckedChange={switchFn} />);

    const button = screen.getByRole("switch");
    expect(button).toBeChecked();

    await waitFor(() => userEvent.click(button));
    expect(button).not.toBeChecked();

    expect(switchFn).toHaveBeenCalled();
  });

  it("should be disabled", async () => {
    const checkFn = vi.fn();

    render(
      <Switch.Button disabled defaultChecked={false} onCheckedChange={checkFn} />
    );

    const button = screen.getByRole("switch");
    await waitFor(() => userEvent.click(button));

    expect(button).toBeDisabled();
    expect(checkFn).not.toHaveBeenCalled();
  });

  it("renders the switch with title", () => {
    const component = render(
      <Switch.Root>
        <Switch.Button />
        <div className="ml-2 flex flex-col">
          <Switch.Label>O Brasil é caramelo</Switch.Label>
        </div>
      </Switch.Root>
    );
    const labelElement = screen.getByText("O Brasil é caramelo");
    expect(labelElement).toBeInTheDocument();
    const switchElement = screen.getByRole("switch");
    expect(switchElement).toBeInTheDocument();
    expect(component.baseElement).toMatchSnapshot();
  });

  it("renders the switch with helper text", () => {
    const component = render(
      <Switch.Root>
        <Switch.Button id="1" />
        <div className="ml-2 flex flex-col">
          <Switch.Label htmlFor="1">O Brasil é caramelo</Switch.Label>
          <HelperText.Label>Texto auxiliar ao Switch</HelperText.Label>
        </div>
      </Switch.Root>
    );
    const labelElement = screen.getByText("O Brasil é caramelo");
    expect(labelElement).toBeInTheDocument();
    const helperTextElement = screen.getByText("Texto auxiliar ao Switch");
    expect(helperTextElement).toBeInTheDocument();
    const switchElement = screen.getByRole("switch");
    expect(switchElement).toBeInTheDocument();
    expect(component.baseElement).toMatchSnapshot();
  });
});
