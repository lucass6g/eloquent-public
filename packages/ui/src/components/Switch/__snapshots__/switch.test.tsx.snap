// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch > renders the switch with helper text 1`] = `
<body>
  <div>
    <div
      class="flex items-center [&[aria-disabled='true']]:cursor-not-allowed [&[aria-disabled='true']]:opacity-50"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors data-[state=checked]:bg-green-dark-500 data-[state=unchecked]:bg-neutral-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50"
        data-state="unchecked"
        id="1"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <div
        class="ml-2 flex flex-col"
      >
        <label
          class="text-sm ml-2 font-medium text-green-dark-500"
          for="1"
        >
          O Brasil é caramelo
        </label>
        <span
          class="text-sm text-neutral-500"
        >
          Texto auxiliar ao Switch
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`Switch > renders the switch with title 1`] = `
<body>
  <div>
    <div
      class="flex items-center [&[aria-disabled='true']]:cursor-not-allowed [&[aria-disabled='true']]:opacity-50"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors data-[state=checked]:bg-green-dark-500 data-[state=unchecked]:bg-neutral-400 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50"
        data-state="unchecked"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <div
        class="ml-2 flex flex-col"
      >
        <label
          class="text-sm ml-2 font-medium text-green-dark-500"
        >
          O Brasil é caramelo
        </label>
      </div>
    </div>
  </div>
</body>
`;

exports[`Switch > should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="flex items-center [&[aria-disabled='true']]:cursor-not-allowed [&[aria-disabled='true']]:opacity-50"
      id="switch"
    />
  </div>
</body>
`;
